File: TimeFlies/vc/todo/TodoDetailController.m
//
//  TodoDetailController.m
//  testTabbarController
//
//  Created by 林伟池 on 15/10/26.
//  Copyright © 2015年 林伟池. All rights reserved.
//

#import "TodoDetailController.h"
#import "UIViewController+TimeFliesNavigationItem.h"
#import "TodoModel.h"
#import "DoneModel.h"

@interface TodoDetailController ()

@end

@implementation TodoDetailController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    if (self.index >= 0) {
        Thing* item = [[TodoModel instance] getThingByIndex:self.index];
        if (item && item.text) {
            self.myText.text = item.text;
        }
    }
//    [self.myText becomeFirstResponder];
    [self customView];
    
//    CIImage* image;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)customView {
//    [self lySetupRightItem];
}
#pragma mark - view init

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
}

- (void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
}
#pragma mark - ui

- (IBAction)onSave:(id)sender
{
    if (self.index != -1) {
        [[TodoModel instance] modifyTodoByIndex:self.index Text:self.myText.text];
    }
    else{
        [[TodoModel instance] addNewTodo:self.myText.text];
    }
    
   
    [self.navigationController popViewControllerAnimated:YES];
}

- (IBAction)onDone:(id)sender
{
    if (![[DoneModel instance] getDoneCounts]) {
        
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"固定到时间轴", nil) message:NSLocalizedString(@"固定后可在时间轴查看，同时不再可修改", nil) preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"否", nil) style:UIAlertActionStyleDefault handler:nil];
        UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"是", nil) style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
            [self onSure];
        }];
        [alertController addAction:okAction];
        [alertController addAction:cancelAction];
        
        [self presentViewController:alertController animated:YES completion:nil];
    }
    else{
        [self onSure];
    }

}

- (void)onSure
{
    if (self.index != -1) {
        [[TodoModel instance] deleteTodoByIndex:self.index];
    }
    
    [[DoneModel instance] addNewPassByText:self.myText.text Type:pass_type_thing];
    
    
    [self.navigationController popViewControllerAnimated:YES];

}

- (IBAction)onDelte:(id)sender
{
    if (self.index != -1) {
        [[TodoModel instance] deleteTodoByIndex:self.index];
    }
    
    [self.navigationController popViewControllerAnimated:YES];
}
#pragma mark - delegate

#pragma mark - notify

@end

File: TimeFlies/vc/todo/TodoDetailController.swift
//
//  TodoDetailController.swift
//  TimeFlies
//
//  Created by 林伟池 on 16/4/8.
//  Copyright © 2016年 林伟池. All rights reserved.
//

import Foundation

class SwiftTodoDetailController: UIViewController {
    
    
    //MARK: property
    @IBOutlet var myText:UITextView?
    var myIndex:Int = 0
    
    //MARK: init
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if myIndex >= 0 {
            let item:Thing? = TodoModel.instance().getThingBy(myIndex)
            if (item != nil) {
                myText?.text = item?.text
            }
        }
        
        customView()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    deinit {
        print("dealloc \(self)")
    }
    
    //MARK: ui
    
    func customView() {
        
    }
    
    @IBAction func onSave(_ sender: NSObject) {
        if myIndex != -1 {
            TodoModel.instance().modifyTodo(by: myIndex, text: myText?.text)
        }
        else {
            TodoModel.instance().addNewTodo(myText?.text)
        }
        
        navigationController?.popViewController(animated: true)
    }
    
    @IBAction func onDone(_ sender: NSObject) {
        if DoneModel.instance().getDoneCounts() > 0 {
            let controller = UIAlertController(title: NSLocalizedString("固定到时间轴", comment: lyCommentDefault), message: NSLocalizedString("固定后可在时间轴查看，同时不再可修改", comment: lyCommentDefault), preferredStyle: .alert)
            
            let cancelAction = UIAlertAction(title: NSLocalizedString("否", comment: lyCommentDefault), style: .default, handler: nil)
            let okAction = UIAlertAction(title: NSLocalizedString("是", comment: lyCommentDefault), style: .default, handler: { [unowned self](action) -> Void in
                self.onSure()
            })
            
            controller.addAction(okAction)
            controller.addAction(cancelAction)
            
            present(controller, animated: true, completion: nil)
        }
        else {
            onSure()
        }
    }
    
    func onSure() {
        if myIndex != -1 {
            TodoModel.instance().deleteTodo(by: myIndex)
        }
        DoneModel.instance().addNewPass(byText: myText?.text, type: .pass_type_thing)
        
        navigationController?.popViewController(animated: true)
    }
    
    @IBAction func onDelete(_ sender: NSObject) {
        if myIndex != -1 {
            TodoModel.instance().deleteTodo(by: myIndex)
        }
        
        navigationController?.popViewController(animated: true)
    }
    //MARK: delegate
    
    
}
